File: t/PACSFramework/VB/StorageServerManager/UI/ServerSettings/Gateway/Control/ItemsGridView.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace Leadtools.Demos.StorageServer.UI
   Partial Friend Class ItemsGridView : Inherits UserControl
      Public Sub New()
         InitializeComponent()

         CanAdd = True

         UpdateToolStripButtons()

         AddHandler ItemsDataGridView.SelectionChanged, AddressOf ItemsDataGridView_SelectionChanged

         AddHandler AddItemToolStrip.Click, AddressOf AddItemToolStrip_Click
         AddHandler ModifyToolStripButton.Click, AddressOf ModifyToolStripButton_Click
         AddHandler DeleteToolStripButton.Click, AddressOf DeleteToolStripButton_Click

         AddHandler ItemsDataGridView.CellDoubleClick, AddressOf ItemsDataGridView_CellDoubleClick
      End Sub

      Public Event SelectedItemChanged As EventHandler

      Public ReadOnly Property SelectedRow() As DataGridViewRow
         Get
            If ItemsDataGridView.SelectedRows.Count > 0 Then
               Return ItemsDataGridView.SelectedRows(0)
            Else
               Return Nothing
            End If
         End Get
      End Property

      Public Property CanAdd() As Boolean
         Get
            Return AddItemToolStrip.Enabled
         End Get
         Set(ByVal value As Boolean)
            AddItemToolStrip.Enabled = value
         End Set
      End Property

      Private Sub ItemsDataGridView_CellDoubleClick(ByVal sender As Object, ByVal e As DataGridViewCellEventArgs)
         If ItemsDataGridView.SelectedRows.Count > 0 Then
            OnModifyItem()
         End If
      End Sub

      Public Property Title() As String
         Get
            Return ContainerGroupBox.Text
         End Get

         Set(ByVal value As String)
            ContainerGroupBox.Text = value
         End Set
      End Property

      Private Sub ItemsDataGridView_SelectionChanged(ByVal sender As Object, ByVal e As EventArgs)
         UpdateToolStripButtons()

         If Not Nothing Is SelectedItemChangedEvent Then
            RaiseEvent SelectedItemChanged(Me, e)
         End If
      End Sub

      Private Sub UpdateToolStripButtons()
         If ItemsDataGridView.SelectedRows.Count > 0 Then
            AddItemToolStrip.Enabled = CanAdd
            DeleteToolStripButton.Enabled = True
            ModifyToolStripButton.Enabled = True
         Else
            AddItemToolStrip.Enabled = CanAdd
            DeleteToolStripButton.Enabled = False
            ModifyToolStripButton.Enabled = False
         End If
      End Sub

      Public Property DataSource() As Object
         Get
            Return ItemsDataGridView.DataSource
         End Get

         Set(ByVal value As Object)
            ItemsDataGridView.DataSource = value

            UpdateToolStripButtons()
         End Set
      End Property

      Public Property DataMember() As String
         Get
            Return ItemsDataGridView.DataMember
         End Get

         Set(ByVal value As String)
            ItemsDataGridView.DataMember = value

            UpdateToolStripButtons()
         End Set
      End Property

      Public Property AddText() As String
         Get
            Return AddItemToolStrip.Text
         End Get

         Set(ByVal value As String)
            AddItemToolStrip.Text = value
         End Set
      End Property

      Public Property ModifyText() As String
         Get
            Return ModifyToolStripButton.Text
         End Get

         Set(ByVal value As String)
            ModifyToolStripButton.Text = value
         End Set
      End Property

      Public Property DeleteText() As String
         Get
            Return DeleteToolStripButton.Text
         End Get

         Set(ByVal value As String)
            DeleteToolStripButton.Text = value
         End Set
      End Property

      Public Event AddItem As EventHandler
      Public Event ModifyItem As EventHandler
      Public Event DeleteItem As EventHandler

      Private Sub OnAddItem()
         If Not Nothing Is AddItemEvent Then
            RaiseEvent AddItem(Me, EventArgs.Empty)
         End If
      End Sub

      Private Sub OnModifyItem()
         If Not Nothing Is ModifyItemEvent Then
            RaiseEvent ModifyItem(Me, EventArgs.Empty)
         End If
      End Sub

      Private Sub OnDeleteItem()
         If Not Nothing Is DeleteItemEvent Then
            RaiseEvent DeleteItem(Me, EventArgs.Empty)
         End If
      End Sub

      Private Sub AddItemToolStrip_Click(ByVal sender As Object, ByVal e As EventArgs)
         OnAddItem()
      End Sub

      Private Sub ModifyToolStripButton_Click(ByVal sender As Object, ByVal e As EventArgs)
         OnModifyItem()
      End Sub


      Private Sub DeleteToolStripButton_Click(ByVal sender As Object, ByVal e As EventArgs)
         OnDeleteItem()
      End Sub
   End Class
End Namespace

File: t/PACSFramework/CS/StorageServerManager/UI/ServerSettings/Gateway/Control/ItemsGridView.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace Leadtools.Demos.StorageServer.UI
{
   public partial class ItemsGridView : UserControl
   {
      public ItemsGridView()
      {
         InitializeComponent();

         CanAdd = true ;
         
         UpdateToolStripButtons ( ) ;
         
         ItemsDataGridView.SelectionChanged += new EventHandler ( ItemsDataGridView_SelectionChanged ) ;

         AddItemToolStrip.Click      += new EventHandler ( AddItemToolStrip_Click ) ;
         ModifyToolStripButton.Click += new EventHandler ( ModifyToolStripButton_Click ) ;
         DeleteToolStripButton.Click += new EventHandler ( DeleteToolStripButton_Click ) ;

         ItemsDataGridView.CellDoubleClick += new DataGridViewCellEventHandler ( ItemsDataGridView_CellDoubleClick ) ;
      }

      public event EventHandler SelectedItemChanged ;
      
      public DataGridViewRow SelectedRow
      {
         get
         {
            if ( ItemsDataGridView.SelectedRows.Count > 0 )
            {
               return ItemsDataGridView.SelectedRows [ 0 ] ;
            }
            else
            {
               return null ;
            }
         }
      }
      
      public bool CanAdd
      {
         get 
         {
            return AddItemToolStrip.Enabled ;
         }
         set 
         {
            AddItemToolStrip.Enabled  = value ;
         }
      }
      
      void ItemsDataGridView_CellDoubleClick ( object sender, DataGridViewCellEventArgs e )
      {
         if ( ItemsDataGridView.SelectedRows.Count > 0 ) 
         {
            OnModifyItem ( ) ;
         }
      }

      public string Title
      {
         get
         {
            return ContainerGroupBox.Text ;
         }
         
         set
         {
            ContainerGroupBox.Text = value ;
         }
      }
      
      void ItemsDataGridView_SelectionChanged ( object sender, EventArgs e )
      {
         UpdateToolStripButtons ( ) ;
         
         if ( null != SelectedItemChanged ) 
         {
            SelectedItemChanged ( this, e ) ;
         }
      }

      private void UpdateToolStripButtons ( )
      {
         if ( ItemsDataGridView.SelectedRows.Count > 0 )
         {
            AddItemToolStrip.Enabled      = CanAdd ;
            DeleteToolStripButton.Enabled = true ;
            ModifyToolStripButton.Enabled = true ;
         }
         else
         {
            AddItemToolStrip.Enabled      = CanAdd ;
            DeleteToolStripButton.Enabled = false ;
            ModifyToolStripButton.Enabled = false ;
         }
      }

      public object DataSource 
      {
         get
         {
            return ItemsDataGridView.DataSource ;
         }
         
         set
         {
            ItemsDataGridView.DataSource = value ;
            
            UpdateToolStripButtons ( ) ;
         }
      }
      
      public string DataMember
      {
         get
         {
            return ItemsDataGridView.DataMember ;
         }
         
         set
         {
            ItemsDataGridView.DataMember = value ;
            
            UpdateToolStripButtons ( ) ;
         }
      }
      
      public string AddText
      {
         get
         {
            return AddItemToolStrip.Text ;
         }
         
         set
         {
            AddItemToolStrip.Text = value ;
         }
      }
      
      public string ModifyText
      {
         get
         {
            return ModifyToolStripButton.Text ;
         }
         
         set
         {
            ModifyToolStripButton.Text = value ;
         }
      }
      
      public string DeleteText
      {
         get
         {
            return DeleteToolStripButton.Text ;
         }
         
         set
         {
            DeleteToolStripButton.Text = value ;
         }
      }

      public event EventHandler AddItem ;
      public event EventHandler ModifyItem ;
      public event EventHandler DeleteItem ;

      private void OnAddItem()
      {
         if ( null != AddItem ) 
         {
            AddItem ( this, EventArgs.Empty ) ;
         }
      }

      private void OnModifyItem()
      {
         if ( null != ModifyItem ) 
         {
            ModifyItem ( this, EventArgs.Empty ) ;
         }
      }

      private void OnDeleteItem()
      {
         if ( null != DeleteItem ) 
         {
            DeleteItem ( this, EventArgs.Empty ) ;
         }
      }

      void AddItemToolStrip_Click(object sender, EventArgs e)
      {
         OnAddItem ( ) ;
      }
      
      void ModifyToolStripButton_Click(object sender, EventArgs e)
      {
         OnModifyItem ( ) ;
      }

      
      void DeleteToolStripButton_Click(object sender, EventArgs e)
      {
         OnDeleteItem ( ) ;
      }
   }
}
